components/ca-resource.html

```diff
@@ -0,0 +1,123 @@
+         * <dt>ca-resource-ready</dt><dd>Fired when the resource has been fetched successfully. {@link value} will be available.
+         * <dt>ca-resource-unauthorized</dt><dd>Fired when the resource fetch failed with 401 Unauthorised because the user's credentials could not be verified.</dd>
+         * <dt>ca-resource-forbidden</dt><dd>Fired when the resource fetch failed with 403 Forbidden because the user's credentials were not acceptable to retrieve it.</dd>
+         * </dl>
+         * TODO: make this able to PUT resources back by changing the 'value' property
+         */
+        class ApiResourceElement extends HTMLElement {
+            /**
+             * observedAttributes - get a list of attributes that should effect re-fetch.
+             * @return {Array} list of attributes that effect a re-fetch.
+             */
+            static get observedAttributes() {
+                return ['href'];
+            }
+
+            /**
+             * @description Executes when any attribute is changed on the element
+             * @type {Event}
+             * @param {string} attrName - the name of the attribute to have changed
+             * @param {string} oldVal - the old value of the attribute
+             * @param {string} newVal - the new value of the attribute
+             * @returns {undefined}
+             */
+            attributeChangedCallback(attrName, oldVal, newVal) {
+                if (ApiResourceElement.observedAttributes.indexOf(attrName) >= 0) {
+                    this.fetch();
+                }
+            }
+
+            /**
+             * @returns {string} URI to which this resource refers
+             */
+            get href() {
+                return this.getAttribute('href');
+            }
+
+            /**
+             * @param {string} value, URI to which this resource refers
+             */
+            set href(value) {
+                this.setAttribute('href', value);
+            }
+
+            /**
+             * @returns {object|string|Blob} value of fetching the resource, as the best type available (e.g. object for JSON, string for text, or failing that a BLOB)
+             */
+            get value() {
+                if (this._decodedResponse) {
+                    return this._decodedResponse;
+                }
+                return undefined;
+            }
+
+            attachedCallback() {
+                if (this.href) {
+                    this.fetch();
+                }
+            }
+
+            /**
+             * Fetches the resource at the specified {@link href}.
+             * @return {Promise<Response>|null} promise of a Fetch response, or null if the fetch was cancelled
+             */
+            fetch() {
+                if (!document.contains(this)) return null; // Don't fetch if we're not attached to a document.
+
+                this._fetchResponse = this._decodedResponse = undefined;
```

Comment:
Wasn't there a lint rule to error on this type assignment?

Reply:
Nope